### ISSUE TITLE
Make VAAPI connection type configurable

### ISSUE THREAD

**Issue by vredez** [FreeRDP/FreeRDP#6666]
**TLDR:** This is necessary to get hardware acceleration working (again) with NVIDIA graphics cards.

FreeRDP uses FFmpeg for VAAPI hardware acceleration. If not specified, the FFmpeg API internally prefers the DRM connection type over the X11 one. Unfortunately, the VAAPI driver for NVIDIA users (libva-vdpau-driver), which seems to be unmaintained since 2012, only supports the X11 connection type.

**Describe the solution you'd like**
It would be really nice if there would be a command line option to specify if DRM or X11 shall be used, e.g.:
`--vaapi-connection-mode=[drm|x11]`

**Additional context**
From a technical point of view, the setting could be directly passed to the `aw_hwdevice_ctx_create(...)` call in `h264_freerdp.c:539`, which currently does not make use of the `opts` parameter.

**Comment by hardening** [comment]
Well, the ffmpeg is so generic that we could probably add an option to do the decoding using VDPAU or CUDA when running on NVidia hardware.

**Comment by vredez** [comment]
> Well, the ffmpeg is so generic that we could probably add an option to do the decoding using VDPAU or CUDA when running on NVidia hardware.

This would be perfect in the long run, but seems to be a lot of effort. I created a PR for the suggested improvement: [https://github.com/FreeRDP/FreeRDP/pull/6667](url)

**Comment by zamroni111** [comment]
> **TLDR:** This is necessary to get hardware acceleration working (again) with NVIDIA graphics cards.
> 
> FreeRDP uses FFmpeg for VAAPI hardware acceleration. If not specified, the FFmpeg API internally prefers the DRM connection type over the X11 one. Unfortunately, the VAAPI driver for NVIDIA users (libva-vdpau-driver), which seems to be unmaintained since 2012, only supports the X11 connection type.
> 
> **Describe the solution you'd like**
> It would be really nice if there would be a command line option to specify if DRM or X11 shall be used, e.g.:
> `--vaapi-connection-mode=[drm|x11]`
> 
> **Additional context**
> From a technical point of view, the setting could be directly passed to the `aw_hwdevice_ctx_create(...)` call in `h264_freerdp.c:539`, which currently does not make use of the `opts` parameter.

Is this for encoding in server side or decoding in client side?
